Contributor Docs: Add a Default Breaking Change section (#25339)
hashicorp/terraform-provider-azurerm: contributing/topics/guide-new-fields-to-resource.md

* In cases where a new property or block requires additional setup or pre-requisites it makes sense to create a dedicated test for it.

> * Adding a new property or updating an existing property with a default should be checked properly against the current version of Terraform State and the Azure api as tests won't catch if that will cause a breaking change, see our [section on defaults and breaking changes](defaults-and-breaking-changes.md)

Reviewer: ```suggestion
* Adding a new property or updating an existing property with a default should be checked properly against the current version of Terraform State and the Azure API as tests won't be able to catch a potential breaking change, see our [section on defaults and breaking changes](defaults-and-breaking-changes.md)
```
